c_src/membrane_ffmpeg_swscale_plugin/pixel_format_converter/converter.c

```diff
@@ -0,0 +1,122 @@
+#include "converter.h"
+#include <libavutil/error.h>
+#include <libavutil/frame.h>
+#include <libavutil/imgutils.h>
+#include <libswscale/swscale.h>
+
+enum AVPixelFormat string_to_AVPixelFormat(char *);
+
+UNIFEX_TERM create(UnifexEnv *env, uint64_t width, uint64_t height,
+                   char *old_format, char *new_format) {
+  UNIFEX_TERM res;
+  const enum AVPixelFormat input_fmt = string_to_AVPixelFormat(old_format),
+                           output_fmt = string_to_AVPixelFormat(new_format);
+
+  if (input_fmt == AV_PIX_FMT_NONE) {
+    return create_result_error(env, "unknown_input_format");
+  } else if (!sws_isSupportedInput(input_fmt)) {
+    return create_result_error(env, "unsupported_input_format");
+  }
+
+  if (output_fmt == AV_PIX_FMT_NONE) {
+    return create_result_error(env, "unknown_output_format");
+  } else if (!sws_isSupportedOutput(output_fmt)) {
+    return create_result_error(env, "unsupported_output_format");
+  }
+
+  State *state = unifex_alloc_state(env);
+  state->sws_context =
+      sws_getContext(width, height, input_fmt, width, height, output_fmt,
+                     SWS_BICUBIC, NULL, NULL, NULL);
```

Comment:
It would be nice to have this configurable, but I guess it doesn't have to be done right now. Please add a TODO note here and make in issue for it